bt-core/src/main/java/bt/data/digest/Digester.java

```diff
@@ -29,22 +29,20 @@
 
     /**
      * Calculates hash of a data range.
-     *
+     * <p>
      * Important: If some of the data can't be read immediately
      * (e.g. due to idiosyncrasies of underlying storage),
      * then this method may return incorrect result.
      *
      * @return Hash (depends on the algorithm being used)
-     *
      * @since 1.2
      */
     byte[] digest(DataRange data);
 
     /**
-     * Calculates hash of a data range.
+     * Calculates hash of a data range. Fails if any storage unit does not work as expected.
```

Comment:
I think this remark can be a bit more specific about the failure mode :)

Reply:
Yes, it can, and should. Will fix (: